lib/providers/search/search.dart

```diff
@@ -61,253 +72,34 @@ class Search with ChangeNotifier {
     }
   }
 
-  Future searchPosts({@required String query}) async {
+  Future searchPosts({@required String authToken, @required String query, BuildContext context}) async {
```

Comment:
`authToken` is received via `authProvider` in line 86. 
```suggestion
  Future searchPosts({@required String query, BuildContext context}) async {
```